src/scaling/scaling.jl

```diff
@@ -0,0 +1,53 @@
+type ScaledInterpolation{T,N,ITPT,IT,GT,RT} <: AbstractInterpolationWrapper{T,N,ITPT,IT,GT}
+    itp::ITPT
+    ranges::RT
+end
+ScaledInterpolation{T,ITPT,IT,GT,RT}(::Type{T}, N, itp::ITPT, ::Type{IT}, ::Type{GT}, ranges::RT) =
+    ScaledInterpolation{T,N,ITPT,IT,GT,RT}(itp, ranges)
+function scale{T,N,IT,GT}(itp::AbstractInterpolation{T,N,IT,GT}, ranges::Range...)
+    length(ranges) == N || error("Must scale $N-dimensional interpolation object with exactly $N ranges (you used $(length(ranges)))")
+    for d in 1:N
+        iextract(IT,d) != NoInterp || ranges[d] == 1:size(itp,d) || error("NoInterp dimension $d must be scaled with unit range 1:$(size(itp,d))")
+    end
+
+    ScaledInterpolation(T,N,itp,IT,GT,ranges)
+end
+
+@generated function getindex{T,N,ITPT,IT<:DimSpec}(sitp::ScaledInterpolation{T,N,ITPT,IT}, xs...)
+    length(xs) == N || error("Must index into $N-dimensional scaled interpolation object with exactly $N indices (you used $(length(xs))")
+    if IT <: Tuple
+        # handle dimspecs
+        interp_indices = map(it -> IT.parameters[it] != NoInterp, 1:length(IT.parameters))
+        :(getindex(sitp.itp, map(coordlookup, $interp_indices, sitp.ranges, xs)...))
+    else
+        :(getindex(sitp.itp, coordlookup(sitp.ranges, tuple(xs...))...))
+    end
+end
+
+size(sitp::ScaledInterpolation, d) = size(sitp.itp, d)
+
+coordlookup(r::FloatRange, x) = (r.divisor * x - r.start) ./ r.step + one(eltype(r))
+coordlookup(r::StepRange, x) = (x - r.start) ./ r.step + one(eltype(r))
+coordlookup(r::UnitRange, x) = x - r.start + one(eltype(r))
+coordlookup(i::Bool, r::Range, x) = i ? coordlookup(r, x) : x
+coordlookup{N}(r::NTuple{N}, x::NTuple{N}) = map(coordlookup,r,x)
+coordlookup{N}(i::NTuple{N}, r::NTuple{N}, x::NTuple{N}) = map(coordlookup, i, r, x)
```

Comment:
I'm seeing a type instability in (some of) these methods that I can't pin down and explain ([reproduction script](https://gist.github.com/tlycken/18db1ed677cd9074b8fe)). It seems that Julia is unable to infer the type even of `coordlookup(2:2.3:10, 5.1)`, which is super-weird, since the quantities involved in calculations in that case are `Float64`.

```
julia> @code_warntype(Interpolations.coordlookup(2:2.3:10, 4.1))
Variables:
  r::FloatRange{Float64}
  x::Float64

Body:
  begin  # C:\Users\Tomas Lycken\.julia\v0.4\Interpolations\src\scaling/scaling.jl, line 33:
      return (Base.box)(Base.Float64,(Base.add_float)((Base.box)(Base.Float64,(Base.div_float)((Base.box)(Base.Float64,(
Base.sub_float)((Base.box)(Base.Float64,(Base.mul_float)((top(getfield))(r::FloatRange{Float64},:divisor)::Float64,x::Fl
oat64)::Any)::Float64,(top(getfield))(r::FloatRange{Float64},:start)::Float64)::Any)::Float64,(top(getfield))(r::FloatRa
nge{Float64},:step)::Float64)::Any)::Float64,(Base.box)(Float64,(Base.sitofp)(Float64,1)::Any)::Float64)::Any)::Float64
  end::Float64
```

Note that there's a bunch of boxing operations as well as `Any` inferred. I have no idea what to do about this.